lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -228,6 +252,25 @@ def message_from(action, response)
         response['status_detail'] || response['message']
       end
 
+      def success_from_get(response)
+        if response.instance_of? Array
+          return true
+        end
+        return false unless response['status_code']
+
+        %w[100 200 400 600].include? response['status_code'].to_s
+      end
+
+      def message_from_get(response)
+        if success_from_get(response)
+          'OK'
+        elsif !(response.instance_of? Array) && response['error']
+          response['error']['description']
+        elsif !(response.instance_of? Array) && response['error_description']
+          response['error_description']
+        end
```

Comment:
The check here seems to only work for the `GET /payment-methods` API and not any other GET requests to local. Can we make it more generic to cover all GET API calls?

Reply:
Addressed.